frontend/src/hooks/food/useFilterSelectionDisplay.ts

```diff
@@ -0,0 +1,33 @@
+import { KEYWORD_EN } from '@/constants/food';
+import { useFoodFilterContext } from '@/context/food';
+import { generateQueryString } from '@/router/routes';
+import { KeywordEn } from '@/types/food/client';
+
+import useEasyNavigate from '../@common/useEasyNavigate';
+import useValidQueryString from '../common/useValidQueryString';
+
+export const useFilterSelectionDisplay = () => {
+  const { replaceQueryString } = useEasyNavigate();
+  const { toggleFilter } = useFoodFilterContext();
+  const filterListQueryString = useValidQueryString(KEYWORD_EN);
+
+  const removeFilter = (category: KeywordEn, value: string) => {
+    let newFilterString;
+
+    if (filterListQueryString[category]?.includes(`,${value}`)) {
+      newFilterString = filterListQueryString[category]?.replace(`,${value}`, '');
+    } else if (filterListQueryString[category]?.includes(`${value},`)) {
+      newFilterString = filterListQueryString[category]?.replace(`${value},`, '');
+    } else newFilterString = filterListQueryString[category]?.replace(value, '');
+
+    const newQueryString = generateQueryString({
+      ...filterListQueryString,
+      [category]: newFilterString,
+    });
+
+    toggleFilter(category, value);
+    replaceQueryString(newQueryString, { exclude: [] });
```

Comment:
```suggestion
const filterList = filterListQueryString[category]?.split(',');

    if (!filterList) return;

    if (filterList.includes(value)) {
      filterList.splice(filterList.indexOf(value), 1);

      const updatedQueryString = filterList.join(',');

      const newQueryString = generateQueryString({
        ...filterListQueryString,
        [category]: updatedQueryString,
      });

      replaceQueryString(newQueryString, { exclude: [] });
    }
```

`else... if`문을 개선해 보았어요!